script.js

```diff
@@ -0,0 +1,179 @@
+    const error = validateConfiguration(packageJson);
+    if (error && error.length > 0) {
+        console.error("Invalid package.json: " + error);
+        return;
+    }
+
+    // We have validated the config. It exists in all its glory
+    let binName = packageJson.goBinary;
+    let binNameUnused = binName + ".exe";
+    const binPath = packageJson.directories.bin;
+
+    // Binary name on Windows has .exe suffix
+    if (process.platform === "win32") {
+        binNameUnused = binName;
+        binName += ".exe";
+    }
+
+    return {
+        binName,
+        binNameUnused,
+        binPath
+    };
+}
+
+/**
+ * Reads the configuration from application's package.json,
+ * validates properties, copies the binary from the package and stores at
+ * ./bin in the package's root. NPM already has support to install binary files
+ * specific locations when invoked with "npm install -g"
+ *
+ *  See: https://docs.npmjs.com/files/package.json#bin
+ */
+
+function install(callback) {
+    const opts = parsePackageJson();
+    if (!opts) {
+        return callback("Invalid inputs");
+    }
+
+    const source = path.join(__dirname, 'dist', `enonic_${PLATFORM_MAPPING[process.platform]}_${ARCH_MAPPING[process.arch]}`, opts.binName);
+
+    if (!fs.existsSync(source)) {
+        console.error(`Looking for ${opts.binName} at "${source}": Not found.`);
+        return;
+    }
+
+    const targetPath = path.join(__dirname, opts.binPath);
+    const target = path.join(targetPath, opts.binName);
+
+    console.log(`Copying the relevant binary for platform "${process.platform}"`);
+    fs.copyFileSync(source, target, fs.constants.COPYFILE_FICLONE);
+}
+
+function preinstall(callback) {
+    const opts = parsePackageJson();
+    if (!opts) {
+        return callback("Invalid inputs");
+    }
+
+    const targetPath = path.join(__dirname, opts.binPath);
+    const unusedBinary = path.join(targetPath, opts.binNameUnused);
+
+    console.log(`Deleting unused binary: ${unusedBinary}`);
+
+    try {
+        fs.unlinkSync(unusedBinary);
+    } catch(err) {
+        //
+    }
+}
+
+function setReleaseVersion() {
+    const { execSync } = require("child_process");
+    const version = execSync("git describe --tags --abbrev=0").toString().slice(1).trim();
+
+    execSync(`awk -v ver="${version}" '{gsub(/"version": ".*"/,"\\"version\\": \\""ver"\\"")}1' package.json > tmp && mv tmp package.json`);
+}
+
+function prepublish(callback) {
+    const opts = parsePackageJson();
+    if (!opts) {
+        return callback("Invalid inputs");
+    }
+
+    console.log('Executing prepublish script...');
+    const targetPath = path.join(__dirname, opts.binPath);
+
+    if (!fs.existsSync(targetPath)) {
+        fs.mkdirSync(targetPath);
+    }
+
+    const file1 = path.join(targetPath, opts.binName);
+    const file2 = path.join(targetPath, opts.binNameUnused);
+
+    fs.writeFileSync( file1, '' );
+    fs.writeFileSync( file2, '' );
```

Comment:
Probably don't need a whitespace between brackets and parameters, since we use different style in this file.